index.js

```diff
@@ -78,7 +77,15 @@ class PreloadPlugin {
         }).forEach(entry => {
           entry = `${publicPath}${entry}`;
           if (options.rel === 'preload') {
-            filesToInclude+= `<link rel="${options.rel}" href="${entry}" as="${options.as}">\n`;
+            // If `as` value is not provided in option, dynamically determine the correct
+            // value depends on suffix of filename. Otherwise use the given `as` value.
+            let asValue;
+            if (!options.as) {
+              asValue = entry.match(/\.css$/) ? 'style' : 'script';
```

Comment:
Agreed! However, previously I didn't put that into code for two reasons:
1. I noticed that files transferred using file-loader are not put into chunks. In this way, it seems that there might be very rare situation where other types will be processed in plugin. (could be wrong, as I am not Webpack specialist)
2. Also, it will be a long list, if we would like to use suffix to handle different files. (and easy to miss some)

I think I can add another feature based on this change:
if `as` in config is provided as a function callback, it will be used to generate `as` attribute in preload. I am thinking of a signature looks like: `function (entryName: string): string`, where the return value is `as` attribute which will be used for this entry. In this way, it should be easy for users to extend the functionality, while for common usage, the plugin already covers.

If you think this solution works, I will add another commit for it. Let me know your thoughts.
:-)